libcxx/src/experimental/tzdb.cpp

```diff
@@ -94,14 +96,23 @@ static void __skip(istream& __input, string_view __suffix) {
 }
 
 static void __matches(istream& __input, char __expected) {
-  if (std::tolower(__input.get()) != __expected)
-    std::__throw_runtime_error((string("corrupt tzdb: expected character '") + __expected + '\'').c_str());
+  _LIBCPP_ASSERT_INTERNAL(std::islower(__expected), "lowercase characters only here!");
+  char __c = __input.get();
+  if (std::tolower(__c) != __expected)
```

Comment:
```suggestion
  if (char __c = __input.get(); std::tolower(__c) != __expected)
```
The same for the hunk below. I would suggest to commit these two hunks directly to main as an NFC patch.